src/build-2024/docs/Exercise/02 Add Chat History.md

```diff
@@ -0,0 +1,58 @@
+﻿# Exercise : Add Chat History
+
+This console application demonstrates a basic chat interface using OpenAI's GPT model (such as GPT-3.5-Turbo or GPT-4-Turbo) through the Microsoft Semantic Kernel. The application maintains a stateful chat interface, where it continuously takes user input and provides responses generated by the AI model, while remembering previous context.
+
+1. Open the project you have created in [01 Hello Semantic Kernel](./01%20Hello%20Semantic%20Kernel.md) in VS Code or Visual Studio.
+
+1. Add the following using statment at the top of `Program.cs` file.
+
+    ```csharp
+    using Microsoft.SemanticKernel.ChatCompletion;
+    ```
+
+1. Add the following code next to kernel initialization to get the required service from the kernel which is chat the completion service.
+
+    ```csharp
+    var chatService = kernel.GetRequiredService<IChatCompletionService>();
```

Comment:
To make it easier to follow, consider showing the other code before this line. Perhaps add text that says

```suggestion
Change the code from 
    ```csharp
    var kernel = Kernel.CreateBuilder()
          .AddOpenAIChatCompletion(openAIChatCompletionModelName, Environment.GetEnvironmentVariable("OPENAI_API_KEY"))
          .Build();
    ```
to

    ```csharp
    var kernel = Kernel.CreateBuilder()
          .AddOpenAIChatCompletion(openAIChatCompletionModelName, Environment.GetEnvironmentVariable("OPENAI_API_KEY"))
          .Build();

    var chatService = kernel.GetRequiredService<IChatCompletionService>();
    ```
```

Or something like that? It does get a bit wordy, though. You could also try text like "After the line that starts with `var kernel...` add the following code. Maybe play around with that a little and see what you think?